src/Control/Distributed/Closure.hs

```diff
@@ -52,3 +58,12 @@ import Data.Constraint (Dict(..))
 -- 'Control.Distributed.Closure.TH.withStatic' if it becomes too tedious.
 class c => Static c where
   closureDict :: Closure (Dict c)
+
+-- | A newtype-wrapper useful for defining instances of classes indexed by
+-- higher-kinded types.
+newtype WrappedArrowClosure a b = WrapArrowClosure
+  { unwrapClosureArrow :: Closure (a -> b)
+  } deriving (Binary)
```

Comment:
For consistency with `base`, which defines e.g. `WrappedMonad` and `WrappedArrow` (the latter of which also being present in `profunctors`).